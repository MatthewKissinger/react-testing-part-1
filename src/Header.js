export default function Header( { text } ) {
    return (
        <header>
            <h2 
                className="hello"
                data-testid="h1tag">{text}</h2>
        </header>
    )
}