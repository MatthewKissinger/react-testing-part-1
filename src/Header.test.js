import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Header from './Header'

afterEach(cleanup);

it("renders", () => {
    const {asFragment} = render(<Header text="Hello!" />);
    expect(asFragment()).toMatchSnapshot();
})

it("inserts text in h1", () => {
    render(<Header text="Hello!" />)

    expect(screen.getByTestId('h1tag')).toHaveTextContent("Hello!");
    expect(screen.getByText("Hello!")).toHaveClass("hello");
})





