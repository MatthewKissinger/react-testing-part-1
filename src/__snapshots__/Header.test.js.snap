// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h2
      class="hello"
      data-testid="h1tag"
    >
      Hello!
    </h2>
  </header>
</DocumentFragment>
`;
